torch_glow/src/CachingGraphRunner.cpp

```diff
@@ -20,61 +20,128 @@
 
 #include "glow/Support/Support.h"
 
+#include <torch/csrc/jit/argument_spec.h>
+#include <torch/csrc/utils/hash.h>
+
 namespace glow {
 
-llvm::Error CachingGraphRunner::runGraph(const torch::jit::Node *node,
-                                         torch::jit::Stack &stack) {
-  // If this is the first time this subgraph has been run then create a new
-  // GraphInfo object to store information about it.
+namespace {
+/// Returns a hash that represents a given PT subgraph represented by the PT
+/// node \p node and a set of tensor shape inputs to that subgraph from the \p
+/// stack.
+size_t getGraphHash(const torch::jit::Node *node,
+                    const torch::jit::Stack &stack) {
+  const std::shared_ptr<torch::jit::Graph> graph = node->g(at::attr::Subgraph);
+  const auto &graphInputs = graph->inputs();
+  const auto inputs = torch::jit::last(stack, graphInputs.size());
+  torch::jit::CompleteArgumentSpec spec(/*with_grad*/ false, inputs);
+
+  return torch::hash_combine(reinterpret_cast<size_t>(node), spec.hashCode());
+}
+
+} // namespace
 
+llvm::Expected<CachingGraphRunner::PerGlowGraphInfo *>
+CachingGraphRunner::loadGraphImpl(const torch::jit::Node *node,
+                                  torch::jit::Stack &stack) {
   const std::shared_ptr<torch::jit::Graph> graph = node->g(at::attr::Subgraph);
+  RETURN_ERR_IF_NOT(graph, "A subgraph couldn't be found for this node");
+
   const auto &graphInputs = graph->inputs();
-  const auto numInputs = graphInputs.size();
-  auto inputs = torch::jit::last(stack, numInputs);
-  const char *const functionName = "PyTorchFunction";
+  auto inputs = torch::jit::last(stack, graphInputs.size());
+
+  size_t hash = getGraphHash(node, stack);
```

Comment:
getGraphHash repeats the same code as:
 
  `const std::shared_ptr<torch::jit::Graph> graph = node->g(at::attr::Subgraph);
   auto inputs = torch::jit::last(stack, graphInputs.size());
   `
  May you can provide **node** & **inputs** as input parameters into getGraphHash function?